packages/react-dom/src/__tests__/ReactTestUtilsAct-test.js

```diff
@@ -503,9 +503,13 @@ function runActTests(label, render, unmount, rerender) {
         spyOnDevAndProd(console, 'error');
         // let's try to cheat and spin off a 'thread' with an act call
         (async () => {
-          await act(async () => {
-            await sleep(50);
-          });
+          try {
+            await act(async () => {
+              await sleep(50);
+            });
+          } catch (e) {
+            // suppress ERR_UNHANDLED_REJECTION
+          }
```

Comment:
## before
<img width="937" alt="Screenshot 2022-02-07 at 19 43 56" src="https://user-images.githubusercontent.com/2394070/152863665-f5910a68-5961-4a22-b4b3-818b3e2e9b7c.png">

from CI
<img width="1012" alt="Screenshot 2022-02-07 at 20 08 51" src="https://user-images.githubusercontent.com/2394070/152863739-69a1983a-3fc3-4b3d-8e93-275fc9ea2142.png">


## after

<img width="942" alt="Screenshot 2022-02-07 at 19 48 13" src="https://user-images.githubusercontent.com/2394070/152863680-a30ee016-1b34-42df-9616-08a8ef842e27.png">

Reply:
Catch-all is risky. How can we narrow it down? Also, why is it needed now exactly?

Reply:
can we catch that specific error?

Reply:
Ah, got you!

Reply:
```js
(async () => {
    try {
      await act(async () => {
        await sleep(50);
      });
    } catch (e) {
      // suppress ERR_UNHANDLED_REJECTION from act()
      if (e.message !== 'act(...) is not supported in production builds of React.') {
        throw e
      }
    }
  })();
```

this works,  or  we just guard it with `__DEV__`?

```js
if (__DEV__) {
    (async () => {
        await act(async () => {
          await sleep(50);
        });
    })();
  }
```

Reply:
hmm. there is `// @gate __DEV__` above. shouldn't this be enough?

Reply:
Aha, things becoming interesting.
`// @gate __DEV__` doesn't skip test in non-Dev, it still runs the test be expects it to fail.
https://github.com/facebook/react/blob/main/scripts/jest/setupTests.js#L270-L286

The test is still run 
https://github.com/facebook/react/blob/cd4eb116cf24d5b05107caf67b96c8e0f0cd7dbc/scripts/jest/setupTests.js#L249

and the exception is not caught, trouble occurs.

since the assertions for this test are under `__DEV__` ([link](https://github.com/facebook/react/blob/cd4eb116cf24d5b05107caf67b96c8e0f0cd7dbc/packages/react-dom/src/__tests__/ReactTestUtilsAct-test.js#L516-L525)), I guess we can just use `__DEV__` as well?
<img width="552" alt="Screenshot 2022-02-07 at 21 29 55" src="https://user-images.githubusercontent.com/2394070/152875243-5fc3bda5-e909-498d-a0ae-ee69ce9b10bb.png">

Reply:
sounds reasonable